applications/CompressiblePotentialFlowApplication/custom_processes/define_3d_wake_process.cpp

```diff
@@ -0,0 +1,910 @@
+
+    RecomputeNodalDistancesToWakeOrWingLowerSurface();
+
+    MarkKuttaElements();
+
+    SaveLocalWakeNormalInElements();
+
+    AddWakeNodesToWakeModelPart();
+
+    if(mCountElementsNumber){
+        CountElementsNumber();
+    }
+
+    if(mWriteElementsIdsToFile){
+        WriteElementIdsToFile();
+    }
+}
+
+// This function initializes the variables and removes all of the elements and
+// nodes of the trailing edge submodelpart
+void Define3DWakeProcess::InitializeTrailingEdgeSubModelpart() const
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    if(root_model_part.HasSubModelPart("trailing_edge_elements_model_part"))
+    {
+        // Clearing the variables and elements of the already existing
+        // trailing_edge_sub_model_part
+        ModelPart& trailing_edge_sub_model_part =
+            root_model_part.GetSubModelPart("trailing_edge_elements_model_part");
+
+        for (auto& r_element : trailing_edge_sub_model_part.Elements()){
+            r_element.SetValue(TRAILING_EDGE, false);
+            r_element.SetValue(KUTTA, false);
+            r_element.Reset(STRUCTURE);
+            r_element.Set(TO_ERASE, true);
+        }
+        auto& r_nodes = trailing_edge_sub_model_part.Nodes();
+        VariableUtils().SetFlag(TO_ERASE, false, r_nodes);
+        trailing_edge_sub_model_part.RemoveElements(TO_ERASE);
+        trailing_edge_sub_model_part.RemoveNodes(TO_ERASE);
+    }
+    else{
+        // Creating the trailing_edge_sub_model_part
+        root_model_part.CreateSubModelPart("trailing_edge_elements_model_part");
+    }
+}
+
+// This function initializes the variables and removes all of the elements and
+// nodes of the wake submodelpart
+void Define3DWakeProcess::InitializeWakeSubModelpart() const
+{
+    ModelPart& root_model_part = mrBodyModelPart.GetRootModelPart();
+    if(root_model_part.HasSubModelPart("wake_elements_model_part"))
+    {
+        // Clearing the variables and elements of the already existing
+        // wake_sub_model_part
+        ModelPart& wake_sub_model_part =
+            root_model_part.GetSubModelPart("wake_elements_model_part");
+
+        for (auto& r_element : wake_sub_model_part.Elements()){
+            r_element.SetValue(WAKE, false);
+            r_element.SetValue(WAKE_ELEMENTAL_DISTANCES, ZeroVector(3));
+            r_element.Set(TO_ERASE, true);
+        }
+        auto& r_nodes = wake_sub_model_part.Nodes();
+        VariableUtils().SetFlag(TO_ERASE, false, r_nodes);
+        wake_sub_model_part.RemoveElements(TO_ERASE);
+        wake_sub_model_part.RemoveNodes(TO_ERASE);
+    }
+    else{
+        // Creating the wake_sub_model_part
+        root_model_part.CreateSubModelPart("wake_elements_model_part");
+    }
+}
+
+// This function marks the trailing edge nodes and finds the wingtip nodes
+void Define3DWakeProcess::MarkTrailingEdgeNodesAndFindWingtipNodes()
+{
+    double max_span_position = std::numeric_limits<double>::lowest();
```

Comment:
```suggestion
    double max_span_position = -std::numeric_limits<double>::max();
```
`max_span_position` can be negative right? If im not mistaken this can be problematic if the wing is not centered at 0,0,0

Reply:
Isn't std::numeric_limits<double>::lowest() = -std::numeric_limits<double>::max()?

Reply:
True sorry, ignore my comment! I was thinking it was an epsilon.